Content.Server/Power/Components/BatterySelfRechargerComponent.cs

```diff
@@ -6,8 +6,29 @@ namespace Content.Server.Power.Components
     [RegisterComponent]
     public sealed partial class BatterySelfRechargerComponent : Component
     {
+        /// <summary>
+        /// Does the entity auto recharge?
+        /// </summary>
         [ViewVariables(VVAccess.ReadWrite)] [DataField("autoRecharge")] public bool AutoRecharge { get; set; }
 
+        /// <summary>
+        /// At what rate does the entity automatically recharge?
+        /// </summary>
         [ViewVariables(VVAccess.ReadWrite)] [DataField("autoRechargeRate")] public float AutoRechargeRate { get; set; }
+
+        /// <summary>
+        /// Should this entity stop automatically recharging if a charge is used?
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)] [DataField("autoRechargePause")] public bool AutoRechargePause { get; set; } = false;
+
+        /// <summary>
+        /// How long should the entity stop automatically recharging if a charge is used?
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)] [DataField("autoRechargePauseTime")] public float AutoRechargePauseTime { get; set; } = 0f;
+
```

Comment:
1. Would you ever add the component without autorecharge? Setting it to false as default also doesn't make sense.
2. Having both autorechargecooldown and autorechargepausetime overcomplicates it, better to just have one.
3. Please don't use datafield strings or properties unless necessary.

Reply:
1. autoRecharge is a field that existed from before this PR.
2. Cooldown is the CURRENT cooldown, so ill remove the datafield string from it.
3. Tied into 2.

Reply:
Reworked it to use a timespan.

Reply:
I mean autorechargepaused.

Also datafields still needed cleaning up.